securedrop/source_user.py

```diff
@@ -66,23 +67,41 @@ class SourceDesignationCollisionError(Exception):
 def create_source_user(
     db_session: Session,
     source_passphrase: "DicewarePassphrase",
-    source_app_crypto_util: "CryptoUtil",
     source_app_storage: "Storage",
 ) -> SourceUser:
     # Derive the source's info from their passphrase
     scrypt_manager = _SourceScryptManager.get_default()
     filesystem_id = scrypt_manager.derive_source_filesystem_id(source_passphrase)
     gpg_secret = scrypt_manager.derive_source_gpg_secret(source_passphrase)
 
-    # Create a designation for the source
-    try:
-        # TODO: Move display_id() out of CryptoUtil
-        journalist_designation = source_app_crypto_util.display_id()
-    except ValueError:
+    # Create a unique journalist designation for the source
+    # TODO: Add unique=True to models.Source.journalist_designation to enforce uniqueness
+    #  as the logic below has a race condition (time we check VS time when we add to the DB)
```

Comment:
I left a TODO here to not have a to do a DB migration. The race condition may also not be a "real" problem.